pigeon/notifications.dart

```diff
@@ -33,6 +34,15 @@ class NotificationChannel {
 
   final String? name;
   final bool? lightsEnabled;
+
+  /// The name of the sound file resource.
+  ///
+  /// The resource ID is retrieved using
+  /// `android.content.res.Resources.getIdentifier`, which is then used to
+  /// generate an `android.resource://` URI. This URI is then passed
+  /// to the `NotificationChannelCompat` builder.
```

Comment:
We could rephrase this to something similar to https://github.com/zulip/zulip-flutter/blob/cdbe14122eac145afaa86d19f152b5451e3c457b/pigeon/notifications.dart#L172-L175, mentioning how `soundResourceName` is expected to be known in keep.xml.